### ISSUE TITLE
Unable to advertise one device to another if the same account restored with different whisper ids

### ISSUE THREAD

**Issue by churik** [status-im/status-react#8715]
### Description

*Type*: Bug
*Summary*: for all new users after merging https://github.com/status-im/status-react/pull/8601 from the same seed phrase are restored different whisper ids, which makes impossible to pair devices.

#### Expected behavior
can advertise one device to another

#### Actual behavior
nothing happens

### Reproduction

- Device A: create account
- Device B: restore the same account
- Device A: open Devices > try to advertise Device A to Device B
- Device B: check devices

### Additional Information

[comment]: # (Please do your best to fill this out.)
- Status version: nightly after merging https://github.com/status-im/status-react/pull/8601
- Operating System: Android, iOS

**Comment by churik** [comment]
@cammellos created an issue to track progress and discussion...
cc @rachelhamlin @annadanchenko

**Comment by cammellos** [comment]
@gravityblast what is the expected behavior? My understanding is that although wallet  key and whisper keys are different if I import the same seed phrase I should get the same wallet key & whisper key, for example:

```
device-1-whisper-key == device-2-whisper-key 
&& 
device-1-wallet-key == device-2-wallet-key
&&
device-1-wallet-key != device-1-whisper-key
```

**Comment by asemiankevich** [comment]
@flexsurfer

**Comment by flexsurfer** [comment]
My understanding is the same as @cammellos, for one seed phrase we should always have the same whisper and default wallet keys

**Comment by guylouis** [comment]
+1
The default whisper key is on a fixed path (m/43'/60'/1581'/0'/0') so always the same adress when we recover a given seed.
Same thing for the default wallet key (m/44'/60'/0'/0/0)

**Comment by flexsurfer** [comment]
@annadanchenko @churik  this is expected `for all new users after merging #8601 from the same seed phrase are restored different whisper ids`, this is was the main goal of multiaccount, to make whisper id different, so if you recover old account there should be same wallet address but different whisper id

**Comment by flexsurfer** [comment]
@churik  or do you mean you install fresh build with  #8601 , **create new account** and get new mnemonic, and recover account on another device with this new mnemonic you have different whisper id ?

**Comment by flexsurfer** [comment]
oh sorry,multiaccounts recovery isn't implemented in  #8601 , so whisper id should the same as before  #8601

**Comment by flexsurfer** [comment]
but anyway i can't reproduce this, because i have always the same whisper id for one mnemonic on different devices

**Comment by flexsurfer** [comment]
multiaccounts recovery is here https://github.com/status-im/status-react/pull/8687

**Comment by gravityblast** [comment]
>@gravityblast what is the expected behavior? My understanding is that although wallet key and whisper keys are different if I import the same seed phrase I should get the same wallet key & whisper key, for example:
>device-1-whisper-key == device-2-whisper-key 
&& 
>device-1-wallet-key == device-2-wallet-key
&&
>device-1-wallet-key != device-1-whisper-key


@cammellos yes it's correct!

**Comment by annadanchenko** [comment]
I can reproduce same issue (another whisper key is recovered) on Android build 22 of Recovery v1 PR https://github.com/status-im/status-react/pull/8687

Steps:
1. on device 1 with build 22 create an account
2. on device 2 use backup phrase from device 1 to recover account. Once recovered check Contact code in profile. It's expected to be the same on both devices

**Comment by flexsurfer** [comment]
hm maybe something wrong with the mnemonic phrase when we generate a new account

**Comment by cammellos** [comment]
Thanks, I will triage this and timebox it to 3 hours, will let you know by tomorrow what comes up.

**Comment by flexsurfer** [comment]
@cammellos  first we need to make sure this method `MultiAccountGenerateAndDeriveAddresses` returns correct mnemonics for generated accounts

**Comment by annadanchenko** [comment]
I must add that wallet address is the same on device1 and device2, so issue is only for whisper id


you can user this phrase 
![Screenshot_20190812-124610](https://user-images.githubusercontent.com/6584810/62860093-6513f580-bcff-11e9-98a1-62d2fe41a1b4.png)

Expected Whisper id:
![Screenshot_20190812-124624](https://user-images.githubusercontent.com/6584810/62860104-6b09d680-bcff-11e9-97eb-6cb5ebdd709e.png)


Expected Wallet address:
![Screenshot_20190812-124636](https://user-images.githubusercontent.com/6584810/62860130-78bf5c00-bcff-11e9-9cff-e58ff0208a1a.png)

**Comment by flexsurfer** [comment]
ok that means `MultiAccountGenerateAndDeriveAddresses` returns wrong whisper ids

**Comment by churik** [comment]
@flexsurfer 
yes, it is reproducible for `develop` for new accounts only (created after #8601 )
It is mentioned as expected result here: https://github.com/status-im/status-react/pull/8601#issuecomment-516306789

**Comment by flexsurfer** [comment]
we provide this path `"m/43'/60'/1581'/0'/0"` to `MultiAccountGenerateAndDeriveAddresses`

**Comment by flexsurfer** [comment]
@churik @annadanchenko @cammellos I've found the reason
in #8601 we still use the old method to recover, so develop is wrong you shouldn't test it
here #8687 @dmitryn has the wrong implementation, we need to fix it
sorry for wasting your time! i'll comment in #8687 and we'll fix it with @dmitryn 
thanks

**Comment by churik** [comment]
cc @rachelhamlin

**Comment by guylouis** [comment]
can this one be closed now that #8687 is merged ?
